src/main/java/org/sunbird/extension/user/impl/UserProviderRegistryImpl.java

```diff
@@ -0,0 +1,297 @@
+package org.sunbird.extension.user.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.typesafe.config.Config;
+import io.opensaber.registry.client.OpensaberClient;
+import io.opensaber.registry.client.data.RequestData;
+import io.opensaber.registry.client.data.ResponseData;
+import io.opensaber.registry.exception.TransformationException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.extension.user.UserExtension;
+import org.sunbird.extension.util.SunbirdExtensionConstants;
+import org.sunbird.extension.util.SunbirdExtensionUtil;
+import org.sunbird.extension.util.TransformJsonUtil;
+
+/**
+ * User profile extension using Open Saber registry for storing adopter specific custom user details
+ *
+ * @author Jaikumar Soundara Rajan
+ */
+public class UserProviderRegistryImpl implements UserExtension {
+
+  private static ObjectMapper mapper = new ObjectMapper();
+  private static OpensaberClient client;
+  private static Config userEnumsConfig;
+  private static Config userWriteConfig;
+  private static Config userReadConfig;
+
+  static {
+    userEnumsConfig =
+        SunbirdExtensionUtil.loadConfig(SunbirdExtensionConstants.USER_ENUMS_MAPPING_FILE);
+    userWriteConfig =
+        SunbirdExtensionUtil.loadConfig(SunbirdExtensionConstants.USER_WRITE_MAPPING_FILE);
+    userReadConfig =
+        SunbirdExtensionUtil.loadConfig(SunbirdExtensionConstants.USER_READ_MAPPING_FILE);
+    client = SunbirdExtensionUtil.createOpensaberClient();
+  }
+
+  @Override
+  public void create(Map<String, Object> userProfileMap) {
+
+    // hardcoded to teacher till userType enhancement is done
+    userProfileMap.put(SunbirdExtensionConstants.USER_TYPE, "teacher");
```

Comment:
Add a new config param (sunbird_default_user_type) in new service.conf (sunbird-lms-mw / sunbird-lms-service).